### PR TITLE
Wmwragg/mention state menu

### PR DESCRIPTION
**Pull request by wmwragg**
A companion branch vector-im/vector-web:wmwragg/mention-state-menu also needs to be merged.

Signed-off-by: William Wragg wm.wragg@gmail.com

### CODE REVIEW

**Comment by wmwragg** [comment]
I'll do a merge up from develop and resolve the conflicts, then add another commit

**Comment by ara4n** [comment]
woo, this is very pretty - thanks :)

Transplanting comments for the record:

 * specifically, wondering about the gap in the hitmask between the roomtile name and the hoverover context menu <-- happy to wait on this until we have signed-off design on the room tile.
 * and if there are any hoverover states for the context menu itself
 * final feedback is that the the gap between the context menu and the LeftPanel is wrong (there should be a small gap in Trevor's design, but it's inset by a few pixels now)

thanks! will wait until you're happy to merge this obv.

**Comment by wmwragg** [comment]
Did I miss some hoverover states for the context menu? I didn't think there were any, I've only implemented the click event to change the state

**Comment by ara4n** [comment]
This was more a question for @antikewl - whether the design intends there to be hoverover states. So far the designs have been missing hoverovers, but I'm not sure that this is intentional.

**Comment by wmwragg** [comment]
@ara4n I looked at what was involved in doing the CSS re-write over the weekend, and it ended up being rather large, and touched a lot of other modules and their CSS, which touched other modules and their CSS etc... So for this ticket I have left it, as this is focussed on the context menu for notifications. When I do the ticket for the LHS panel restyling, I'll look at doing the CSS refactor then.

So the code is pretty much final, I'm only waiting on Trevor for the designs for collapsed mute state, and long room names, and then I'll mark this as done.

**Comment by ara4n** [comment]
@wmwragg cool. the only thing stopping me from merging this is the question marks hovering over the RoomTile tooltip design itself (and the fact that we haven't wired up the various mute states yet).

I'm a bit scared at the number of open PRs we're starting to accumulate; what are your thoughts on merging this, including the WIP design?  Or should we wait?

**Comment by wmwragg** [comment]
I would prefer to wait for the final designs for the long room name and collapsed mute state before merging this, as I would like a PR to close a ticket, rather than multiple PRs to close a ticket, as that will start to get unmanageable. As far as I can tell, there are only 2 PRs waiting to be merged for the design work ribot is doing (well 4 as there are the matching 2 on the vector-web repo as well, but I view those as the sam PR). I'm hoping Trevor will have me unblocked today.

**Comment by wmwragg** [comment]
@ara4n For the RoomTile tooltip design, are we talking about the collapsed tooltip, or the current uncollapsed long hover tooltip? As we can't style the latter it's browser specific and we have no styling control over it.

**Comment by ara4n** [comment]
uh, i have no idea why i said tooltip there - i think i meant 'question marks hovering over the RoomTile design itself'.  sorry; was written pre-coffee :(

In terms pending PRs: i think we have 2 for the mention state (one for RoomTile; one for the menu) and one for the buttons.  I'll sort the buttons one now, and then happy to consider the two RoomTile ones as effectively same issue.

**Comment by wmwragg** [comment]
I think there is only one for the mention state, the one for the RoomTile is just a branch, I merged it into this branch, so it's in this PR. I needed the stuff from that branch as I did further work waiting for the long name design. When this PR is merged it will close issues #1841 and #1746 from the `vector-web` repo. As far as I am aware these are the outstanding paired PRs:

* For buttons issue vector-im/vector-web#1752
    * PR vector-im/vector-web#1828
    * PR matrix-org/matrix-react-sdk#353
* For the notification state issues vector-im/vector-web#1841 and vector-im/vector-web#1746
    * PR vector-im/vector-web#1900
    * PR matrix-org/matrix-react-sdk#369

**Comment by wmwragg** [comment]
@ara4n New long name design. I've also tweaked the gutters so they correctly align. You'll need to pull the `matrix-react-sdk` repo as well.

File: src/components/views/rooms/RoomTile.js [hunk 1]
```diff
@@ -43,16 +44,48 @@ module.exports = React.createClass({
...
             badgeHover : false,
+            menu: false,
+            areNotifsMuted: areNotifsMuted,
         });
     },
 
+    onAction: function(payload) {
+        switch (payload.action) {
+            case 'notification_change':
+                // Is the notificaion about this room
```
Comment (ara4n) [inline src/components/views/rooms/RoomTile.js]: typo

**Comment by ara4n** [comment]
LGTM - thanks for persevering on the epic; i think the end result was worth it :)

File: src/components/views/rooms/RoomTile.js (hunk 1 above)
Comment (wmwragg) [inline src/components/views/rooms/RoomTile.js]: Well spotted. Fixed

**Pull request merged**